File: app/src/main/java/com/whatsap/statussaver/permissions/AppPermissionInterface.java
package com.whatsap.statussaver.permissions;

import android.app.Activity;

/**
 * Created by Swanand Deshpande on 1/3/18.
 */

public interface AppPermissionInterface {
    boolean requestPermissionList(Activity activity, int[] permissionCode, int requestCode);
    void requestPermissionList(Activity activity, String[] permissionCode, int requestCode);
    void showPermissionDialog(Activity activity, AlertDialogProperties alertDialogProperties, String[] permissionNames);
    void closeDialog();
}

File: app/src/main/java/com/whatsap/statussaver/permissions/AppPermissionInterface.kt
package com.whatsap.statussaver.permissions

import android.app.Activity

/**
 * Created by Swanand Deshpande on 1/3/18.
 */
interface AppPermissionInterface {
    fun requestPermissionList(activity: Activity?, permissionCode: IntArray?, requestCode: Int): Boolean
    fun requestPermissionList(activity: Activity?, permissionCode: Array<String?>?, requestCode: Int)
    fun showPermissionDialog(activity: Activity?, alertDialogProperties: AlertDialogProperties?, permissionNames: Array<String?>?)
    fun closeDialog()
}